Sources/Testing/Parameterization/TypeInfo.swift

```diff
@@ -55,14 +142,55 @@ extension TypeInfo: CustomStringConvertible, CustomDebugStringConvertible {
   }
 
   public var debugDescription: String {
-    qualifiedName
+    fullyQualifiedName
   }
 }
 
 // MARK: - Equatable, Hashable
 
-extension TypeInfo: Hashable {}
+extension TypeInfo: Hashable {
+  public static func ==(lhs: Self, rhs: Self) -> Bool {
+    switch (lhs._kind, rhs._kind) {
+    case let (.type(lhs), .type(rhs)):
+      return lhs == rhs
+    default:
+      return lhs.fullyQualifiedNameComponents == rhs.fullyQualifiedNameComponents
+    }
+  }
+
+  public func hash(into hasher: inout Hasher) {
+    switch _kind {
+    case let .type(type):
+      hasher.combine(ObjectIdentifier(type))
+    case let .nameOnly(fqnComponents, _):
+      hasher.combine(fqnComponents)
+    }
+  }
+}
 
 // MARK: - Codable
 
-extension TypeInfo: Codable {}
+extension TypeInfo: Codable {
+  /// A simplified version of ``TypeInfo`` suitable for encoding and decoding.
+  fileprivate struct EncodedForm {
```

Comment:
Interesting pattern!

Reply:
Useful for when you don't want to deal with encoding/decoding containers directly.